src/System.Device.Gpio/System/Device/Pwm/Channels/UnixPwmChannel.Linux.cs

```diff
@@ -90,18 +91,31 @@ private static int GetPeriodInNanoseconds(int frequency)
         /// Sets the frequency for the channel.
         /// </summary>
         /// <param name="frequency">The frequency in hertz to set.</param>
-        private void SetFrequency(int frequency)
+        private void SetFrequency(int frequency, bool forceUpdateDutyCycle = false)
         {
             if (frequency < 0)
             {
                 throw new ArgumentOutOfRangeException(nameof(frequency), frequency, "Value must not be negative.");
             }
 
+            double oldDutyCycle = _dutyCycle;
+
+            if (forceUpdateDutyCycle || (oldDutyCycle != 0 && frequency > _frequency))
+            {
+                // Since frequency is higher setting frequency now might
+                // cause errors because old duty cycle value (in nanoseconds)
+                // might now exceed maximum possible range
+                DutyCycle = 0;
```

Comment:
I believe this didn't work either. I tried in the past setting duty cycle to 0 but couldn't make it work.

Reply:
I haven't seen any issues locally yet. I'll try different combinations like restarting my PI and stuff to see if anything breaks

Reply:
You're right but this is only the case just after the boot